### ISSUE TITLE
New baggage not getting flushed in the current span with otel

### ISSUE THREAD

**Issue by ff-sdesai** [spring-projects-experimental/spring-cloud-sleuth-otel#151]
I have created a spring-boot micro-service with spring-cloud-sleuth-otel. I have also configured a open-telemetry-collector to forward the logs to datadog. The otel-collector is running in a docker container along with the micro-service.
I am trying to add a new baggage in an aspect (com.uplight.aspect.TestApplicationAspect.java). However, the baggage gets added starting from next new span only (Created using @NewSpan in com.uplight.service.TestServiceImpl). 
This issue can be reproduced using a POC with very few files and is uploaded at https://github.com/ff-sdesai/distributed-tracing-spring. The readme.md has the steps to run the application. You will need to enter a valid datadog API key in the .env file.I will not be able to share the key since the key is specific to my organization. But even if you don't enter the key, you will be able to see on the otel-collector console that the span named `HTTP GET` does not print the newly added baggage.

I also found out that this can be achieved in Brave by creating a bean of ScopeDecorator in following way

```
 @Bean
     ScopeDecorator mdcScopeDecorator() {
     return MDCScopeDecorator.newBuilder().clear().add( SingleCorrelationField.newBuilder( countryCodeField() ).flushOnUpdate().build() ).build();
     }
```
However,  I could not find a way to achieve this when not using brave

**Comment by marcingrzejszczak** [comment]
Can you try the latest version of Sleuth OTel and the Cloud BOM and try again?

**Comment by ff-sdesai** [comment]
@marcingrzejszczak  I have updated following dependencies to latest :-

1.  spring-cloud-dependencies (2022.0.1 from 2022.0.0)
2. spring-cloud-sleuth-otel-dependencies and spring-cloud-sleuth-otel-autoconfigure(1.1.1 from 1.1.0)
3. spring-cloud-starter-sleuth (3.1.6 from 3.1.5)

But still don't see baggage added to currently active span

**Comment by marcingrzejszczak** [comment]
BTW `(2022.0.1 from 2022.0.0)` won't work cause Sleuth is not part of the release train. I'm talking about the latest `2021.x` train

**Comment by ff-sdesai** [comment]
> How are you making the call using the baggae api ?

I have added a Spring Aspect to intercept Rest API calls. In this aspect, I have autowired  `com.springframework.cloud.sleuth.Tracer` and I am adding baggage in following manner:-
`tracer.createBaggage( "testProperty", "test-property1" );`

**Comment by ff-sdesai** [comment]
> BTW `(2022.0.1 from 2022.0.0)` won't work cause Sleuth is not part of the release train. I'm talking about the latest `2021.x` train

I did not get you. Are you saying I should not be using 2022.x versions?

**Comment by marcingrzejszczak** [comment]
Yes, that's what I'm saying. Sleuth is not working with the 2022 train - check tihs doc for migration guide https://github.com/micrometer-metrics/tracing/wiki/Spring-Cloud-Sleuth-3.1-Migration-Guide

**Comment by ff-sdesai** [comment]
> Yes, that's what I'm saying. Sleuth is not working with the 2022 train - check tihs doc for migration guide https://github.com/micrometer-metrics/tracing/wiki/Spring-Cloud-Sleuth-3.1-Migration-Guide

Ok. Let me try again with 2021.0.5

**Comment by ff-sdesai** [comment]
I checked and behavior is same with 2021.0.5 also

**Comment by marcingrzejszczak** [comment]
How are you making the call using the baggae api ?

**Comment by ff-sdesai** [comment]
> How are you making the call using the baggae api ?

I have added a Spring Aspect to intercept Rest API calls. In this aspect, I have autowired `com.springframework.cloud.sleuth.Tracer ` and I am adding baggage in following manner:-
`tracer.createBaggage( "testProperty", "test-property1" );`

**Comment by marcingrzejszczak** [comment]
Here you have example of using baggage https://github.com/spring-cloud-samples/spring-cloud-sleuth-samples/tree/main/baggage-producer and here https://github.com/spring-cloud-samples/spring-cloud-sleuth-samples/tree/main/baggage-consumer . Are you using the API in a similar fashion?

**Comment by ff-sdesai** [comment]
> Here you have example of using baggage https://github.com/spring-cloud-samples/spring-cloud-sleuth-samples/tree/main/baggage-producer and here https://github.com/spring-cloud-samples/spring-cloud-sleuth-samples/tree/main/baggage-consumer . Are you using the API in a similar fashion?

Yes, only difference is I have directly autowired Tracer , but I am calling same method on Tracer instance to create the baggage

**Comment by marcingrzejszczak** [comment]
Hmm can you prepare a very simple scenario like the one in micrometer samples that would replicate the problem?

**Comment by ff-sdesai** [comment]
> Hmm can you prepare a very simple scenario like the one in micrometer samples that would replicate the problem?

@marcingrzejszczak  I have already prepared the scenario with the fewest files possible. I can not remove docker files because those are needed for otel-collector to run and also, there are only 3 Java files, aspect ,controller and service all of which contribute in reproducing this issue.

**Comment by marcingrzejszczak** [comment]
OK I'll look into that ASAP
